tests/study/storage/variantstudy/test_variant_study_service.py

```diff
@@ -239,3 +242,181 @@ def test_generate_task(
+                    params=Mock(
+                        spec=RequestParameters,
+                        user=Mock(spec=JWTUser, id=regular_user.id, impersonator=regular_user.id),
+                    ),
+                )
+
+                variant_study_service.get(variant_list[index])
+
+        variant_study_path = Path(tmp_path).joinpath("internal_studies")
+
+        # Check if everything was correctly initialized
+        assert len(list(variant_study_path.iterdir())) == 3
+
+        for variant in variant_study_path.iterdir():
+            assert variant.is_dir()
+            assert list(variant.iterdir())[0].name == "snapshot"
+
+        # Begin tests
+        # A user without rights cannot clear snapshots
+        with pytest.raises(UserHasNotPermissionError):
+            variant_study_service.clear_all_snapshots(
+                datetime.timedelta(1),
+                params=Mock(
+                    spec=RequestParameters,
+                    user=Mock(spec=JWTUser, id=regular_user.id, is_site_admin=Mock(return_value=False)),
+                ),
+            )
+
+        # At this point, variants was not accessed yet
+        # Thus snapshot directories must exist still
+        for variant in variant_study_path.iterdir():
+            assert variant.is_dir()
+            assert list(variant.iterdir())
+
+        # Simulate access for two old snapshots
+        variant_list[0].last_access = datetime.datetime.utcnow() - datetime.timedelta(days=60)
+        variant_list[1].last_access = datetime.datetime.utcnow() - datetime.timedelta(hours=6)
+
+        # Simulate access for a recent one
+        variant_list[2].last_access = datetime.datetime.utcnow() - datetime.timedelta(hours=1)
+        db.session.commit()
+
+        # Clear old snapshots
+        task_id = variant_study_service.clear_all_snapshots(
+            datetime.timedelta(hours=5),
+            Mock(
+                spec=RequestParameters,
+                user=DEFAULT_ADMIN_USER,
+            ),
+        )
+        variant_study_service.task_service.await_task(task_id)
+
+        # Check if old snapshots was successfully cleared
+        nb_snapshot_dir = 0  # after the for iterations, must equal 1
+        for variant_path in variant_study_path.iterdir():
+            if variant_path.joinpath("snapshot").exists():
+                nb_snapshot_dir += 1
+        assert nb_snapshot_dir == 1
+
+        # Clear most recent snapshots
+        task_id = variant_study_service.clear_all_snapshots(
+            datetime.timedelta(hours=-1),
+            Mock(
+                spec=RequestParameters,
+                user=DEFAULT_ADMIN_USER,
+            ),
+        )
+        variant_study_service.task_service.await_task(task_id)
+
+        # Check if all snapshots was cleared
+        nb_snapshot_dir = 0  # after the for iterations, must equal 0
+        for variant_path in variant_study_path.iterdir():
+            if variant_path.joinpath("snapshot").exists():
```

Comment:
Better way to do it is `assert not variant_path.joinpath("snapshot").exists()` instead of introducing a counter and you can even do it inside one line